### DISCUSSION TITLE
为什么要通过 dart ffi 对接？

### DISCUSSION THREAD

**Discussion by yisar** [openkraken/kraken#169] [Q&A] [answered]
绕过 dart，直接对接 flutter 的 c++ engine 不是更好？

**Comment by andycall** [comment]
1. 会和 Flutter 的版本造成 Break Change，未来很难跟随 Flutter 的升级而升级
2. 无法和现有 Flutter 应用融合渲染，我们希望 Kraken 能够和 Flutter Widget 在同一个 View 上工作。
3. 直接对接 Flutter C++ Engine 需要再把 Dart Framework 所实现的渲染流程都照搬一遍，成本很高，而且 Dart 的开发效率比 C++ 要高出不少

**Comment by wssgcg1213** [comment]
修改 Engine 的维护成本无法 Cover 收益

**Comment by yisar** [comment]
虽然从收益和成本角度，确实不值得
但我总觉得，两个带 GC 的 runtime 总归不是好事，可能短期体现不出来

**Reply by wssgcg1213** [reply]
从前端角度垂直看可能是这样，如果站在 Flutter 开发者的视角可能又会有不同，很多决策是不能两全的，找到合适的场景更好一些

**Comment by wssgcg1213** [comment, marked as answer]
绕过 dart 相当于绕过了整个 Flutter 生态，Kraken 项目的定位是 Flutter 生态的一部分，所以绕过 dart 对接 engine 的事情我们暂时不会做。
